src/Microsoft.AspNetCore.Routing/Matching/DfaMatcherBuilder.cs

```diff
@@ -9,6 +9,33 @@
 
 namespace Microsoft.AspNetCore.Routing.Matching
 {
+    internal ref struct CandidateBuilderState
```

Comment:
I started doing this thing to see what it felt like and quickly felt it was stupid. Validate this is stupid for me and I'll move this state to fields on DfaMatcherBuilder